src/composables/useRenderer/index.ts

```diff
@@ -118,145 +99,141 @@ export interface UseRendererOptions extends WebGLRendererParameters {
+    failIfMajorPerformanceCaveat: toValue(options.failIfMajorPerformanceCaveat)
+  }))
+
+  const renderer = shallowRef<WebGLRenderer>(new WebGLRenderer(webGLRendererConstructorParameters.value))
+
+  // since the properties set via the constructor can't be updated dynamically,
+  // the renderer is recreated once they change
+  watch(webGLRendererConstructorParameters, () => {
+    renderer.value.dispose()
+    renderer.value = new WebGLRenderer(webGLRendererConstructorParameters.value)
+  })
+  const { pixelRatio } = useDevicePixelRatio()
+  const { pause, resume, onLoop } = useRenderLoop()
+
+  watchEffect(() => {
+    renderer.value.setSize(sizes.width.value, sizes.height.value)
+  })
+
+  watchEffect(() => {
+    renderer.value.setPixelRatio(pixelRatio.value)
+  })
 
   const { logError } = useLogger()
-  const { pixelRatio } = useDevicePixelRatio()
-  const { pause, resume } = useRenderLoop()
 
-  const updateRendererSize = () => {
-    if (!renderer.value) {
-      return
+  const getThreeRendererDefaults = () => {
+
+    const plainRenderer = new WebGLRenderer()
+
+    const defaults = {
+
+      shadowMap: {
+        enabled: plainRenderer.shadowMap.enabled,
+        type: plainRenderer.shadowMap.type,
+      },
+      toneMapping: plainRenderer.toneMapping,
+      toneMappingExposure: plainRenderer.toneMappingExposure,
+      outputColorSpace: plainRenderer.outputColorSpace,
+      useLegacyLights: plainRenderer.useLegacyLights
     }
+    plainRenderer.dispose()
 
-    renderer.value.setSize(sizes.width.value, sizes.height.value)
-    renderer.value.setPixelRatio(Math.min(pixelRatio.value, 2))
+    return defaults
   }
 
-  const updateRendererOptions = () => {
-    if (!renderer.value) {
-      return
-    }
+  const threeDefaults = getThreeRendererDefaults()
 
-    const rendererPreset = toValue(preset)
+  watchEffect(() => {
+    const rendererPreset = toValue(options.preset)
 
     if (rendererPreset) {
       if (!(rendererPreset in rendererPresets))
         logError('Renderer Preset must be one of these: ' + Object.keys(rendererPresets).join(', '))
+
       merge(renderer.value, rendererPresets[rendererPreset])
 
-      return
+      return // TODO should not return here -> discuss with team whether presets should be part of cientos
     }
 
-    renderer.value.shadowMap.enabled = toValue(shadows) as boolean
-    renderer.value.shadowMap.type = toValue(shadowMapType) as ShadowMapType
-    renderer.value.toneMapping = (toValue(toneMapping) as ToneMapping) || NoToneMapping
-    renderer.value.toneMappingExposure = toValue(toneMappingExposure) as number
-    // Wating for https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65356/files to be merged
-    renderer.value.outputColorSpace = toValue(outputColorSpace as ColorSpace) || LinearSRGBColorSpace
-    if (clearColor?.value) renderer.value.setClearColor(normalizeColor(toValue(clearColor) as TresColor))
+    const shadows = toValue(options.shadows)
```

Comment:
I thought about this for a while, too. But it turned out that the solutions I came up with made the less readable. The options/props don't always match the properties in the renderer (`shadowMap.enabled`↔️ `shadows` for example). I think we could find a cool solution in pair programming next week ☺️.

Reply:
We decided in a meet to leave the code the way it is.

Reply:
I unresolved this thread again because I changed the code because the behavior concerning the presets had to change according to our poll in discord. @alvarosabu Can you please check the code I added in https://github.com/Tresjs/tres/pull/340/commits/225c828486f50f798d7ad9308a94ba7b080355d1? A second look at it can't hurt.